registry/default/magicui/shiny-button.tsx

```diff
@@ -29,45 +29,43 @@ const animationProps = {
   },
 } as AnimationProps;
 
-interface ShinyButtonProps extends HTMLMotionProps<"button"> {
+interface ShinyButtonProps extends Omit<HTMLMotionProps<"button">, "ref"> {
```

Comment:
can you explain the reasoning for omitting the 'ref' again?

Reply:
I don’t think we need to `omit` it, but the main issue arises when adding a `ref` to the button. The `motion.button` accepts a ref that is a different type from a standard one, causing a type conflict.

Declaring the ref in the interface like `ref?: React.Ref<HTMLButtonElement>;` solves the issue. Essentially, you can either declare the type yourself or omit the ref type from the `HTMLMotionProps` so it doesn’t validate the ref and avoids throwing errors.

Let me know.

Reply:
kk lets go with omitting the ref then

Reply:
```js
interface AnimatedSubscribeButtonProps extends HTMLMotionProps<"button"> {
  subscribeStatus: boolean;
  children: React.ReactNode;
  className?: string;
  ref?: React.Ref<HTMLButtonElement>;
}
```

This could be another solution - to explicitly declare the type.

Reply:
wouldn't this result in the ref overriding issue whenever you forwardref?